packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -111,8 +112,15 @@ export const useExecuteCode: EnsembleActionHook<
                 storage,
                 formatter,
                 env: appContext?.env,
-                navigateScreen: (targetScreen: NavigateScreenAction): void =>
-                  navigateApi(targetScreen, screen, navigate),
+                navigateScreen: (targetScreen: string, data: unknown): void =>
+                  navigateApi(
+                    targetScreen,
+                    data,
+                    screen,
+                    storage,
+                    navigate,
+                    customScope,
```

Comment:
This scope should actually be passed in to the evaluate fn. Your navigate fn just serves as a type of callback, it doesn't do any evaluation itself